apps/passport-client/components/screens/DeviceLoginScreen.tsx

```diff
@@ -0,0 +1,82 @@
+import { Spacer } from "@pcd/passport-ui";
+import {
+  ChangeEvent,
+  FormEvent,
+  useCallback,
+  useContext,
+  useState
+} from "react";
+import styled from "styled-components";
+import { DispatchContext } from "../../src/dispatch";
+import { BigInput, Button, CenterColumn, H2, TextCenter } from "../core";
+import { AppContainer } from "../shared/AppContainer";
+
+export function DeviceLoginScreen() {
+  const [email, setEmail] = useState("");
+  const [secret, setSecret] = useState("");
+  const [state, dispatch] = useContext(DispatchContext);
+
+  const onAuthenticate = useCallback(
+    function (e: FormEvent<HTMLFormElement>) {
+      e.preventDefault();
+      dispatch({
+        type: "new-device-login-passport"
+      });
+      dispatch({
+        type: "device-login",
+        secret: secret,
+        email: email
+      });
+    },
+    [dispatch, email, secret]
+  );
+
+  return (
+    <AppContainer bg="primary">
+      <Container>
+        <Spacer h={64} />
+        <TextCenter>
+          <H2>DEVICE LOGIN</H2>
+          <Spacer h={32} />
+          <TextCenter>
+            Log in using the device-specific email and secret key
+          </TextCenter>
```

Comment:
we should probably add instructions for how to do this somewhere, we can do that later. added an issue to follow up https://github.com/proofcarryingdata/zupass/issues/398